Frontend/ui-driver/src/Screens/DriverReferralScreen/View.purs

```diff
@@ -303,31 +560,59 @@ appQRCodeView push state =
             , margin $ MarginVertical 10 7
             , color Color.black800
             ] <> FontStyle.h2 TypoGraphy
-          , imageView
-            [ width $ V 280
-            , height $ V 280
-            , gravity CENTER
-            , imageWithFallback $ fetchImage FF_ASSET "ny_driver_app_qr_code"
-            ]
+          , appQRCode "ny_driver_app_qr_code" push state true
           , PrimaryButton.view (push <<< PrimaryButtonActionController state) (primaryButtonConfig state)
          ]
     ]
 
+appQRCode :: forall w . String -> (Action -> Effect Unit) -> DriverReferralScreenState -> Boolean -> PrestoDOM (Effect Unit) w
+appQRCode image push state isqrvisible = 
+    linearLayout
+      [ width MATCH_PARENT
+      , height WRAP_CONTENT
+      , gravity CENTER
+      , visibility if isqrvisible then VISIBLE else GONE
+      ][imageView
+            [ width $ V 280
+            , height $ V 280
+            , imageWithFallback $ fetchImage FF_ASSET image
+            ]
+      ]
+
 qrScreenView :: forall w . (Action -> Effect Unit) -> DriverReferralScreenState -> PrestoDOM (Effect Unit) w
 qrScreenView push state = 
   linearLayout
     [ width MATCH_PARENT
     , height WRAP_CONTENT
     ][ imageView 
-            [ height $ V 0
-            , width $ V 0
-            , id $ getNewIDWithTag "DriverReferralQRScreen"
-            , imageWithFallback $ fetchImage FF_ASSET "ny_driver_app_qr_code"
-            ]
-          , imageView 
-            [ height $ V 0
-            , width $ V 0
-            , id $ getNewIDWithTag "DriverReferralScreenLogo"
-            , imageWithFallback $ fetchImage FF_ASSET "ic_namma_yatri_logo"
-            ]
-    ]
\ No newline at end of file
+          [ height $ V 0
+          , width $ V 0
+          , id $ getNewIDWithTag "DriverReferralQRScreen"
+          , imageWithFallback $ fetchImage FF_ASSET "ny_driver_app_qr_code"
+          ]
+      , imageView
+          [ height $ V 0
+          , width $ V 0
+          , id $ getNewIDWithTag "CustomerReferralQRScreen"
```

Comment:
we can use our existing data type for this case matching on `CustomerReferralQRScreen` and `DriverReferralQRScreen` and also get/set `id` using `show` instance on them